onnxruntime/core/providers/cuda/generator/constant_of_shape.cc

```diff
@@ -21,26 +21,26 @@ ONNX_OPERATOR_KERNEL_EX(
         .TypeConstraint("T2", DataTypeImpl::AllFixedSizeTensorTypes()),
     ConstantOfShape);
 
-Status ConstantOfShape::Compute(OpKernelContext* ctx) const {
+Status ConstantOfShape::ComputeInternal(OpKernelContext* ctx) const {
   Tensor* output_tensor = nullptr;
   ORT_RETURN_IF_ERROR(PrepareCompute(ctx, &output_tensor));
   auto output_data = output_tensor->MutableDataRaw();
   const auto size = output_tensor->Shape().Size();
   const void* value_ptr = GetValuePtr();
   const auto element_size = output_tensor->DataType()->Size();
+
+#define CASE(TYPE)                                                                                          \
+  case sizeof(TYPE):                                                                                        \
+    if (size > 0) {                                                                                         \
```

Comment:
How about put if (size > 0) out of switch so that it is not needed to check it in every case.

Reply:
the logic is not identical. it is expected to fail if type is not supported.